tasks/Task.cpp

```diff
@@ -13,37 +18,638 @@ Task::~Task()
+        Arena::SetNodeValue<int64_t>(m_device->GetTLDeviceNodeMap(),
+            "CcpSwitchoverKey",
+            m_switchover_key);
+        Arena::SetNodeValue<GenICam::gcstring>(m_device->GetTLDeviceNodeMap(),
+            "DeviceAccessStatus",
+            "ReadWrite");
+
+        // confirm if operation was successfull
+        device_access_status =
+            Arena::GetNodeValue<GenICam::gcstring>(m_device->GetTLDeviceNodeMap(),
+                "DeviceAccessStatus");
+        if (device_access_status == "ReadWrite") {
+            LOG_INFO_S << "Task retrieved the Camera Read/Write Access." << endl;
+            return true;
+        }
+    }
+    LOG_WARN_S << "Task failed to retrieved the Camera Read/Write Access." << endl;
+    return false;
+}
+
+bool Task::configureCamera()
+{
+    LOG_INFO_S << "Performing Factory Reset" << endl;
+    if (!factoryReset())
+        return false;
+
+    LOG_INFO_S << "Configuring camera." << endl;
+    if (!acquisitionConfiguration())
+        return false;
+
+    LOG_INFO_S << "Setting StreamBufferHandlingMode." << endl;
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetTLStreamNodeMap(),
+        "StreamBufferHandlingMode",
+        "NewestOnly");
+
+    LOG_INFO_S << "Setting StreamAutoNegotiatePacketSize." << endl;
+    Arena::SetNodeValue<bool>(m_device->GetTLStreamNodeMap(),
+        "StreamAutoNegotiatePacketSize",
+        true);
+
+    LOG_INFO_S << "Setting StreamPacketResendEnable." << endl;
+    Arena::SetNodeValue<bool>(m_device->GetTLStreamNodeMap(),
+        "StreamPacketResendEnable",
+        true);
+
+    LOG_INFO_S << "Setting PixelFormat." << endl;
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetNodeMap(),
+        "PixelFormat",
+        convertFrameModeToPixelFormat(_camera_format.get(), _depth.get()).c_str());
+
+    // When horizontal binning is used, horizontal decimation (if supported) is not
+    // available. When vertical binning is used, vertical decimation (if supported) is not
+    // available.
+
+    // If the camera is acquiring images, AcquisitionStop must be called before adjusting
+    // binning settings.
+    if (!binningConfiguration())
+        return false;
+
+    if (!decimationConfiguration())
+        return false;
+
+    if (!dimensionsConfiguration())
+        return false;
+
+    if (!exposureConfiguration())
+        return false;
+
+    Frame* frame =
+        new Frame(_width.get(), _height.get(), _depth.get(), _camera_format.get(), 0, 0);
+    m_frame.reset(frame);
+
+    return true;
+}
+
+void Task::startCamera()
+{
+    LOG_INFO_S << "Starting frame acquisition." << endl;
+    m_device->StartStream();
+}
+
+void Task::acquireFrame()
+{
+    try {
+        Arena::IImage* image = m_device->GetImage(_frame_timeout.get().toMilliseconds());
+        if (image->IsIncomplete()) {
+            LOG_ERROR_S << "Image is not complete!! " << endl;
+            m_device->RequeueBuffer(image);
+        }
```

Comment:
Use this as an if guard, returning for here so you can remove the `else` block (not its contents)